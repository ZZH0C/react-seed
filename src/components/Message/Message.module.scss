.message_section_container {
  position: absolute;
  width: 100%;
  padding-top: 3rem;
  display: flex;
  align-items: center;
  justify-content: space-around;
  flex-direction: column;
}

.message_section {
  display: flex;
  flex-direction: column;
  padding: 1rem;
  width: 90%;
  box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
  overflow-wrap: break-word;
}

.message_section_text {
  margin: 0.5rem 0;
}

.message_section_iframe {
  height: 60rem;
  width: 100%;
  border: none;
}
