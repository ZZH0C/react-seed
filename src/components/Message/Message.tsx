import React from 'react';
import styles from './Message.module.scss';
import classNames from 'classnames';
import { ButtonLink } from '../ButtonLink/ButtonLink';
import { useLocation } from 'react-router-dom';

interface MessageProps extends React.HTMLAttributes<HTMLElement> {
  from: string;
  title: string;
  text: string;
}

export const Message: React.FC<MessageProps> = ({
  from,
  title,
  text,
  children,
}) => {
  const { search } = useLocation();
  const realHref = {
    pathname: '/home',
    search,
  };

  return (
    <section className={`mainSection ${styles.main_section}`}>
      {children}
      <div className={classNames(styles.message_section)}>
        <ButtonLink href={realHref}>Return</ButtonLink>
        <span className={classNames(styles.message_section_text)}>
          From: {from}
        </span>
        <span className={classNames(styles.message_section_text)}>
          Subject: {title}
        </span>
        <span className={classNames(styles.message_section_text)}>
          <iframe
            seamless
            className={classNames(styles.message_section_iframe)}
            title={'Selected message'}
            src={`data:text/html;charset=UTF-8 ;base64,${text}`}
          />
        </span>
      </div>
    </section>
  );
};
