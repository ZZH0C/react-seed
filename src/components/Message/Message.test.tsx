import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { Message } from './Message';

describe('components/ButtonLink', () => {
  const mockProps = {
    from: 'SOME_USER',
    title: 'SOME_TITLE',
    text: 'some text',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(
      shallow(
        <Message
          from={mockProps.from}
          title={mockProps.title}
          text={mockProps.text}
        >
          {mockProps.text}
        </Message>,
      ),
    ).toMatchSnapshot();
  });
  it('should render without text and match snapshot', () => {
    expect(
      shallow(
        <Message
          from={mockProps.from}
          title={mockProps.title}
          text={mockProps.text}
        />,
      ),
    ).toMatchSnapshot();
  });
});
