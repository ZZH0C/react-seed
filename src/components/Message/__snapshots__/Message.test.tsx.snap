// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonLink should render and match snapshot 1`] = `
<section
  className="message_section_container"
>
  <div
    className="message_section"
  >
    <span
      className="message_section_text"
    >
      From: 
      SOME_USER
    </span>
    <span
      className="message_section_text"
    >
      Subject: 
      SOME_TITLE
    </span>
    <span
      className="message_section_text"
    >
      some text
    </span>
  </div>
</section>
`;

exports[`components/ButtonLink should render without children and match snapshot 1`] = `
<section
  className="message_section_container"
>
  <div
    className="message_section"
  >
    <span
      className="message_section_text"
    >
      From: 
      SOME_USER
    </span>
    <span
      className="message_section_text"
    >
      Subject: 
      SOME_TITLE
    </span>
    <span
      className="message_section_text"
    />
  </div>
</section>
`;
